clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -270,12 +262,12 @@ public boolean partitionsAutoAssigned() {
         return !this.subscription.isEmpty();
     }
 
-    public void consumed(TopicPartition tp, long offset) {
-        assignedState(tp).consumed(offset);
+    public void position(TopicPartition tp, long offset) {
```

Comment:
So, it seems like we could have also solved this problem by fixing the places where the `fetched` and `consumed` get into inconsistent states. I like that we've simplified things, but I'm wondering if this also means we're ruling out (or making more complicated) possible improvements (e.g., prefetching larger chunks of data by allowing fetch requests even when `consumed < fetched`) or if such changes would require effectively reverting the bulk of this patch.